root/usr/lib64/ruby/vendor_ruby/discovery/screen/foreman.rb

```diff
@@ -0,0 +1,46 @@
+def screen_foreman mac = nil, gw = nil, proxy_url = cmdline('proxy.url'), proxy_type = cmdline('proxy.type')
+  Newt::Screen.centered_window(59, 20, "Foreman credentials")
+  f = Newt::Form.new
+  t_desc = Newt::Textbox.new(2, 2, 54, 6, Newt::FLAG_WRAP)
+  t_desc.set_text "Provide full URL (http(s)://host:PORT) to the Server or Proxy " +
+  "according to the type selected. Ports are usually 8443, 8448 or 9090 according " +
+  "to configuration."
+  l_url = Newt::Label.new(2, 8, "Foreman URL:")
+  l_type = Newt::Label.new(2, 10, "Connection type:")
+  t_url = Newt::Entry.new(20, 8, "", 36, Newt::FLAG_SCROLL)
+  r_server = Newt::RadioButton.new(20, 10, "Server", 0, nil)
+  r_proxy = Newt::RadioButton.new(32, 10, "Proxy", 1, r_server)
+  b_ok = Newt::Button.new(34, 15, "Next")
+  b_cancel = Newt::Button.new(46, 15, "Cancel")
+  if gw.nil?
+    proxy_url ||= ''
+  else
+    proxy_url ||= "https://#{gw}:8443"
```

Comment:
Can the default just be 443? The most common case is probably directly the Foreman.  And 8443 doesn't work for Katello, so that might be confusing for users.  Let them enter the proxy port, if they want to connect to one